pymongo/errors.py

```diff
@@ -183,15 +198,15 @@ def __init__(
         self,
         error: str,
         code: Optional[int] = None,
-        details: Optional[Mapping[str, Any]] = None,
+        details: Optional[_DocumentOut] = None,
         max_wire_version: Optional[int] = None,
     ) -> None:
         error_labels = None
         if details is not None:
             error_labels = details.get("errorLabels")
         super().__init__(_format_detailed_error(error, details), error_labels=error_labels)
         self.__code = code
-        self.__details = details
+        self.__details = details or {}
```

Comment:
I'm concerned about this change. Technically it's breaking since details used to default to `None` but now will be `{}`.

Reply:
Hm, that's a fair concern. What if I undo the changes in this file and then add type ignores for now?

Reply:
Yes that sounds safer.

Reply:
Sounds good, just changed it back